provider/openstack/openstack.go

```diff
@@ -21,61 +21,101 @@ import (
 	"github.com/gophercloud/gophercloud/pagination"
 )
 
+// OpenStackComputeManager provides OpenStack Nova API interaction.
+type OpenStackComputeManager interface {
+	// ListFlavors returns list of flavors matching disk and mem criteria.
+	// Args:
+	// - minDiskGB   ... only list flavors with disk >= minDiskGB
+	// - minMemoryMB ... only list flavors with mem >= minMemoryMB
+	ListFlavors(int, int) []flavors.Flavor
```

Comment:
The reason I am asking for this is because in this case we can have only a single interface in case we ever want to add new providers.